ProcessList.c

```diff
@@ -170,73 +320,96 @@ static long ProcessList_treeProcessCompare(const void* v1, const void* v2) {
    const Process *p1 = (const Process*)v1;
    const Process *p2 = (const Process*)v2;
 
-   return p1->pid - p2->pid;
+   // return p1->tree_left - p2->tree_left;
+   return (p1->tree_left < p2->tree_left ? -1 : 1);
 }
 
-void ProcessList_sort(ProcessList* this) {
-   if (!this->settings->treeView) {
-      Vector_insertionSort(this->processes);
-   } else {
-      // Save settings
-      int direction = this->settings->direction;
-      // Sort by PID
-      Vector_quickSortCustomCompare(this->processes, ProcessList_treeProcessCompare);
-      int vsize = Vector_size(this->processes);
-      // Find all processes whose parent is not visible
-      int size;
-      while ((size = Vector_size(this->processes))) {
-         int i;
-         for (i = 0; i < size; i++) {
-            Process* process = (Process*)(Vector_get(this->processes, i));
-            // Immediately consume not shown processes
-            if (!process->show) {
-               process = (Process*)(Vector_take(this->processes, i));
-               process->indent = 0;
-               Vector_add(this->processes2, process);
-               ProcessList_buildTree(this, process->pid, 0, 0, direction, false);
-               break;
-            }
-            pid_t ppid = Process_getParentPid(process);
-            // Bisect the process vector to find parent
-            int l = 0, r = size;
-            // If PID corresponds with PPID (e.g. "kernel_task" (PID:0, PPID:0)
-            // on Mac OS X 10.11.6) cancel bisecting and regard this process as
-            // root.
-            if (process->pid == ppid)
-               r = 0;
-
-            while (l < r) {
-               int c = (l + r) / 2;
-               pid_t pid = ((Process*)(Vector_get(this->processes, c)))->pid;
-               if (ppid == pid) {
-                  break;
-               } else if (ppid < pid) {
-                  r = c;
-               } else {
-                  l = c + 1;
-               }
-            }
-            // If parent not found, then construct the tree with this root
-            if (l >= r) {
-               process = (Process*)(Vector_take(this->processes, i));
-               process->indent = 0;
-               Vector_add(this->processes2, process);
-               ProcessList_buildTree(this, process->pid, 0, 0, direction, process->showChildren);
+static long ProcessList_treeProcessCompareByPID(const void* v1, const void* v2) {
+   const Process *p1 = (const Process*)v1;
+   const Process *p2 = (const Process*)v2;
+
+   // return p1->tree_left - p2->tree_left;
+   return (p1->pid - p2->pid);
```

Comment:
dito?

Reply:
What it actually means?
I am not very educated in these acronyms :(

Reply:
dito means "the same as before". In this case: to use the `SPACESHIP_NUMBER` macro as with the previous functions.